src/Microsoft.DocAsCode.Build.Engine/LinkPhaseHandler.cs

```diff
@@ -197,16 +197,13 @@ private void RegisterXRefSpec(SaveResult result)
                     if (Context.XRefSpecMap.TryGetValue(spec.Uid, out XRefSpec xref))
                     {
                         Logger.LogWarning(
-                            $"Uid({spec.Uid}) has already been defined in {((RelativePath)xref.Href).RemoveWorkingFolder()}.",
-                            null,
-                            null,
-                            null,
-                            WarningCodes.Build.DuplicateUids);
-                    }
-                    else
-                    {
-                        Context.RegisterInternalXrefSpec(spec);
+                           $"Uid({spec?.Uid}) has already been defined in {((RelativePath)spec?.Href).RemoveWorkingFolder()}.",
+                           null,
+                           null,
+                           null,
+                           WarningCodes.Build.DuplicateUids);
                     }
+                    Context.RegisterInternalXrefSpec(spec);
```

Comment:
do we always need to register here?

Reply:
If it's duplicated uid, go into this method to check if the spec will be record in final xref map or not.
If it's new uid, keep same as before.

Reply:
Why change `xref.Href` to `spec?.Href`?

We can merge `TryGetValue+LogWarning` into `AddOrUpdate` in `RegisterInternalXrefSpec`.

minor: use `code: WarningCodes.Build.DuplicateUids`

Reply:
`spec?.Href` is an omission when changing code. Reverted the change.

`merge TryGetValue+LogWarning into AddOrUpdate in RegisterInternalXrefSpec` with this change, we will (log duplicate uids warnings for unchanged models in incremental build)[https://github.com/dotnet/docfx/blob/dev/src/Microsoft.DocAsCode.Build.Engine/LinkPhaseHandlerWithIncremental.cs#L192] as well.